### PR TITLE
Removes Telescience

### PR DESCRIPTION
**Pull request by LordFowl**
I've finally had enough. After a long discussion with staff I think I've reached a decision that most people can agree with. If you can't, I'm sorry. I'm done.

This PR fixes a few bugs with echoes, and adds in loredev requests for the feature. This list may expand, but currently it is:
1: Phoron walls/windows should prevent echoes from locomoting through them (Moon/Paradox)
2: Vaurca should be unable to become echoes and should be unable to host echoes (Bygone)
3: Unathi should be able to absorb echoes to increase their power level (Jackboot)

- [x] Fix punctuation bug
- [x] Remove force message ability
- [x] Code in phoron impediments
- [x] Code in amnesia for echoes.
- [x] Code in anti-gibbing for echoes
- [x] Code in echoes randomly teleporting when first separated.
- [x] Code in clauses preventing Vaurca from interaction with Echoes
- [x] Code in species specific interactions
- [x] Testing

### CODE REVIEW

**Comment by LordFowl** [comment]
!review

**Comment by MarinaGryphon** [comment]
I can't believe you failed to use a different word for bluespace echoes in every line of the changelog. I believe 'spirits' was used twice. Disgraceful.

**Comment by TehFlaminTaco** [comment]
IPCs should be unable to produce or house echos.

**Comment by LordFowl** [comment]
We have determined not to do that.

File: code/modules/mob/living/simple_animal/shade.dm [hunk 1]
```diff
@@ -124,39 +130,93 @@
...
+			var/valid = 0
+			for(var/word2 in words_in_memory)
+				if(lowertext(word1) != lowertext(word2))
+					continue
+				else
+					valid = 1
+					break
+			if(!valid)
+				message = replacetext(message, word1, pick(words_in_memory))
+		message = slur(message,rand(10, 20))
```
Comment (BurgerLUA) [inline code/modules/mob/living/simple_animal/shade.dm]: This is redundant. Just use ```slur(message,15)```.

File: html/changelogs/stuffstuffstuff.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Cosmic Echoes can no longer move through phoron walls/windows."
```
Comment (BurgerLUA) [inline html/changelogs/stuffstuffstuff.yml]: Tbh some of this changelog can be cut down to size.

Comment (LordFowl) [reply html/changelogs/stuffstuffstuff.yml]: Such as?

File: code/modules/mob/living/simple_animal/shade.dm (hunk 1 above)
Comment (LordFowl) [reply code/modules/mob/living/simple_animal/shade.dm]: How so?

Comment (BurgerLUA) [reply code/modules/mob/living/simple_animal/shade.dm]: The second value of proc/slur is already randomized. You're basically calling a rand of a rand with this.

File: html/changelogs/stuffstuffstuff.yml (hunk 2 above)
Comment (BurgerLUA) [reply html/changelogs/stuffstuffstuff.yml]: I think it doesn't need to be on 7 lines. Some of them can be moved into one line.

I suggest putting all the additions on one line, and all the tweaks of previous existing mechanics on one line, and all the bugfixes on another.

File: code/modules/mob/living/simple_animal/shade.dm (hunk 1 above)
Comment (BurgerLUA) [reply code/modules/mob/living/simple_animal/shade.dm]:
```proc/slur(phrase, strength = 100)
	phrase = html_decode(phrase)
	var/leng=lentext(phrase)
	var/counter=lentext(phrase)
	var/newphrase=""
	var/newletter=""
	while(counter>=1)
		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
		if(prob(strength))
			if(rand(1,3)==3)
				if(lowertext(newletter)=="o")	newletter="u"
				if(lowertext(newletter)=="s")	newletter="ch"
				if(lowertext(newletter)=="a")	newletter="ah"
				if(lowertext(newletter)=="c")	newletter="k"
			switch(rand(1,15))
				if(1,3,5,8)	newletter="[lowertext(newletter)]"
				if(2,4,6,15)	newletter="[uppertext(newletter)]"
				if(7)	newletter+="'"
		newphrase+="[newletter]";counter-=1
	return newphrase```

This is proc/slur

File: html/changelogs/stuffstuffstuff.yml (hunk 2 above)
Comment (LordFowl) [reply html/changelogs/stuffstuffstuff.yml]: Multi-line changelogs are preferable to changelogs with several line's worth of information condensed into a single paragraph - namely a changelog should be entries that are line-sized, not paragraph sized. Entries that convey little to no information are the ones that should be culled.

This PR changes a large number of things, so it is only natural that it should have a number of entries.

**Pull request merged**